practices/documentation.md

```diff
@@ -0,0 +1,70 @@
+---
+slug: documentation
+
+stages:
+  - development
+short_description: Documentation in a software project helps the end users to use the software effectively and saves a lot of time for them while trying to troubleshoot issues.
+
+tags:
+- Documentation
+- Helpful instructions
+- Docusaurus
+- GitBook
+
+keywords:
+- Documentation
+- Helpful instructions
+- Docusaurus
+- GitBook
+
+---
+
+# Documentation
+
+**TL;DR**
+
+Every software should have documentation written for it. It helps users by providing instructions on how to use the software. Follow these guidelines to write good documentation.
+
+## What Is Documentation
+
+Documentation in a software project is a manual that describes the architecture of the software and provides helpful instructions to the users. These instructions include an installation guide, how to use the software, and also guides to resolve common issues users might experience while working with the software.
+
+## Why You Might Want Documentation
+
+Providing clear documentation is important because it saves a lot of time for the users, other developers in the team, and even sometimes for the developer who wrote the code. How?
+
+- Users while using the software for the first time, might not know how to navigate within the software or achieve the intended functionality. Clear documentation comes in handy at these times.
+
+- Documentation helps other developers or open source contributors(in case of open source project) who are working in the codebase to understand the code better and makes the development process faster and simpler.
+
+- In case of a situation, where the developer took a break from the project for a brief period or when the codebase becomes larger, it might be hard for the developer to remember all the details without the help of documentation.
+
+## Problems Documentation Helps to Solve
+
+- [Long feedback loops](/problems/long-feedback-loops)
+- [Poor code quality](/problems/poor-code-quality)
+- Less user support
+
```

Comment:
There are more problems that documentation solves. Imagine yourself in a situation that you need to integrate some features, but there is no documentation around it. How would that feel? I hate the frustration when I spend hours on something that feels like a 5-minute task just because there is not enough information (or clear information. or correct information). Do you know that feeling?